torch/csrc/jit/passes/graph_fuser.cpp

```diff
@@ -249,18 +312,48 @@ struct GraphFuser {
     return isFusableOnlyAsExitNode(node);
   }
 
+  bool compatibleDevices(Node * consumer, Value * producer) {
+    auto consumer_device = getDevice(consumer);
+    auto producer_device = getDevice(producer->node());
+
+    if (consumer_device.type() == DeviceType::Unknown ||
+        producer_device.type() == DeviceType::Unknown) {
+      return false;
+    }
+
+    if (consumer_device.type() == DeviceType::CPU ||
+        producer_device.type() == DeviceType::CPU) {
+      if (!sharedFusionCompiler().canCompileOnCPU()) {
+        return false;
+      }
```

Comment:
The inner conditional is just `return sharedFusionCompiler.canCompileOnCPU()`.

Reply:
But it's possible that consumer_device is DeviceType::CPU and producer_device is DeviceType::CUDA. I wanted to check if the fusion compiler can compile on CPU before going on to the other conditional statements later.

This logic is a little convoluted so I'll try to think of a way to clean it up.

Reply:
I'm not sure if I follow. I think if you were to replace the 3 lines above with what I proposed, then it would end up being 100% equivalent. Isn't that the case?

Reply:
Consider the case where consumer_device.Type() == DeviceType::CPU and producer_device.type() == CUDA:

If I changed
```
if (consumer_device.type() == DeviceType::CPU ||
        producer_device.type() == DeviceType::CPU) {
      if (!sharedFusionCompiler().canCompileOnCPU()) {
        return false;
      }
}
```
to 
```
if (consumer_device.type() == DeviceType::CPU ||
        producer_device.type() == DeviceType::CPU) {
      return sharedFusionCompiler.canCompileOnCPU()
   }
}
```
and `sharedFusionCompiler.canCompileOnCPU()` is true, then `compatibleDevices(...)` would return true. However, consumer_device is different from producer_device, so we should not be returning true here

Reply:
Oh god, of course. Never mind.